### PR TITLE
feat: protoc-gen-scala support

### PR DESCRIPTION
**Pull request by strophy**
Initial implementation of `protoc-gen-scala`. Still need to figure out what Java dependencies are required to run this in the target stage, particularly on `arm64`.

### CODE REVIEW

**Review by rvolosatovs — COMMENTED** [review]
What's the status here? Is the `arm64` not supported yet? If so, how about we file an issue about it and fix up in a follow-up?

**Comment by strophy** [comment]
Not ready yet, I need to block out some time for more testing and to get the version strings compatible with Renovate here. I'll test `arm64` build at the same time and see if we can support it immediately or later.

**Comment by strophy** [comment]
I've made progress on this and understand the build process (and the errors I was seeing) much better now. I expect we will be able to complete this PR using a different build method when [this fix is merged](https://github.com/coursier/coursier/pull/2627), it doesn't look like it will take too long.

**Comment by strophy** [comment]
@rvolosatovs this is finally ready. The builder will not be versioned by Renovate since Scala sbt downloads different compilers as necessary. I ended up getting help from the ScalaPB maintainers and using the original build approach, will ask for help sooner next time :sweat_smile:

**Review by rvolosatovs — COMMENTED** [review]
Changes look good to me, but could you drop the merge commit?
Simply squashing all the commits into one based on top of latest `main` would be fine as well IMO (although as a nitpick it would probably be nicer to have a separate commit for reordering of the README list first)

File: Dockerfile [hunk 1]
```diff
@@ -324,6 +325,16 @@ RUN pkg \
         /usr/local/lib/node_modules/ts-protoc-gen
 RUN install -D protoc-gen-ts /out/usr/bin/protoc-gen-ts
 
+FROM sbtscala/scala-sbt:${SCALA_SBT_VERSION} as protoc_gen_scala
+RUN mkdir -p /scala-protobuf
+ARG PROTOC_GEN_SCALA_VERSION
+RUN curl -sSL https://api.github.com/repos/scalapb/ScalaPB/tarball/${PROTOC_GEN_SCALA_VERSION} | tar xz --strip 1 -C /scala-protobuf
+WORKDIR /scala-protobuf
+RUN gu install native-image
+ENV NATIVE_IMAGE_INSTALLED=true
```
Comment (rvolosatovs) [inline Dockerfile]: Is this only relevant at build time or do we also need to include it at runtime? Probably a good idea to add a small comment documenting what this is and why we need it

**Comment by strophy** [comment]
> Changes look good to me, but could you drop the merge commit? Simply squashing all the commits into one based on top of latest `main` would be fine as well IMO (although as a nitpick it would probably be nicer to have a separate commit for reordering of the README list first)

Thanks for the clear instructions, I installed GitLens and finally think I did it correctly! :tada:

**Comment by rvolosatovs** [comment]
Great job, thanks!

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**